### PR TITLE
Fix Gerrit BC pipeline statuses

### PR DESCRIPTION
**Pull request by varsanojidan**
This shows pipeline statuses for Gerrit changes in BC

## Test plan

Manual

image

### CODE REVIEW

File: enterprise/internal/batches/state/state.go [hunk 1]
```diff
@@ -292,6 +294,38 @@ func parseAzureDevOpsBuildState(s adobatches.PullRequestStatusState) btypes.Chan
...
+	return combineCheckStates(states)
+}
+
+func parseGerritBuildState(s string) btypes.ChangesetCheckState {
+	switch s {
+	case "-2", "-1":
+		return btypes.ChangesetCheckStateFailed
+	case "0":
+		return btypes.ChangesetCheckStatePending
+	case "+2", "+1":
```
Comment (varsanojidan) [inline enterprise/internal/batches/state/state.go:322]: For non-Code-Review labels, the numbers can range bw -2 -> +2, depending on how they are declared on the Gerrit side.

File: internal/extsvc/gerrit/types.go [hunk 2]
```diff
@@ -101,24 +101,22 @@ type ChangeReviewComment struct {
...
 //
 //	+2 : approved, can be merged
 //	+1 : approved, but needs additional reviews
 //	 0 : no score
 //	-1 : needs changes
 //	-2 : rejected
-type Approvals struct {
-	CodeReview string `json:"Code-Review"`
-}
+const CodeReviewKey = "Code-Review"
```
Comment (varsanojidan) [inline internal/extsvc/gerrit/types.go:112]: The only label thats guaranteed to exist is the Code-Review label which signifies actual human reviews, all other labels are generally tests and pipelines so I wrap them all into the check state.

File: enterprise/internal/batches/state/state.go [hunk 3]
```diff
@@ -121,8 +121,10 @@ func computeCheckState(c *btypes.Changeset, events ChangesetEvents) btypes.Chang
 		return computeBitbucketCloudBuildState(c.UpdatedAt, m, events)
 	case *azuredevops.AnnotatedPullRequest:
 		return computeAzureDevOpsBuildState(m)
-	case *gerritbatches.AnnotatedChange, *protocol.PerforceChangelistState:
-		// Gerrit and Perforce don't have builds built-in, its better to be explicit by still
+	case *gerritbatches.AnnotatedChange:
+		return computeGerritBuildState(m)
+	case *protocol.PerforceChangelistState:
+		// Perforce don't have builds built-in, its better to be explicit by still
```
Comment (courier-new) [inline enterprise/internal/batches/state/state.go:127]:
```suggestion
		// Perforce doesn't have builds built-in, its better to be explicit by still
```

File: enterprise/internal/batches/state/state.go [hunk 4]
```diff
@@ -292,6 +294,38 @@ func parseAzureDevOpsBuildState(s adobatches.PullRequestStatusState) btypes.Chan
...
+		states = append(states, v)
+	}
+	return combineCheckStates(states)
+}
+
+func parseGerritBuildState(s string) btypes.ChangesetCheckState {
+	switch s {
+	case "-2", "-1":
+		return btypes.ChangesetCheckStateFailed
+	case "0":
```
Comment (courier-new) [inline enterprise/internal/batches/state/state.go:320]: Is this one not meant to be `" 0"`, too?

**Review by courier-new — APPROVED** [review]
Gotta love Gerrit.

**Comment by varsanojidan** [comment]
>Gotta love Gerrit.
Do i really have to?

File: enterprise/internal/batches/state/state.go (hunk 4 above)
Comment (varsanojidan) [reply enterprise/internal/batches/state/state.go]: yep!, I actually fixed it and forgot to push 😅

**Pull request merged**